src/auto-trace.js

```diff
@@ -58,10 +64,15 @@ export function throwAsyncStacktrace(...extraMiddlewares) {
  * @param {function (Object err) => Object newErr } [extraMiddlewares] - middleware function(s) that will be called on rawError before being handled
  * @returns {Error} 
  */
-export function syncStacktrace(rawError, ...extraMiddlewares){
-	const err = rawError || new Error();
-	const middlewareErr = executeMiddleware(err, extraMiddlewares);
-	return wrapObjectWithError(middlewareErr);
+export function syncStacktrace(callback = ()=>{}) {
```

Comment:
syncStacktrace is now a higher order function - this is so that the middleware will work properly